### ISSUE TITLE
Using this?.some throws ReferenceError: _this is not defined

### ISSUE THREAD

**Issue by fstaffa** [vercel/next.js#54054]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.5.0: Thu Jun  8 22:22:20 PDT 2023; root:xnu-8796.121.3~7/RELEASE_ARM64_T6000
    Binaries:
      Node: 18.16.1
      npm: 9.5.1
      Yarn: N/A
      pnpm: N/A
    Relevant Packages:
      next: 13.4.16
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.3
    Next.js Config:
      output: N/A
```

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/fstaffa/next-_this-bug-reproduction

### To Reproduce

Run next dev , and open the browser on http://localhost:3000

Next throws an exception
<img width="977" alt="image" src="https://github.com/vercel/next.js/assets/294522/f8a4b2cb-4fba-41cd-8a4b-74fe6828fb8b">

### Describe the Bug

This behavior started appearing with next 13.4.13 (worked fine on 13.4.12)

The class

```typescript
class Test
{
  test() {
    console.log(this?.some)
  }
}
```

gets transpiled into

```javascript
class Test {
  test() {
    console.log(_this === null || _this === void 0 ? void 0 : _this.some);
  }
}
```
which incorrectly uses _this instead of this (or at least declaring var _this = this).

### Expected Behavior

No error is thrown, compiler either declares __this_ variable or uses _this_ as is

<sub>[NEXT-1518](https://linear.app/vercel/issue/NEXT-1518/using-thissome-throws-referenceerror-this-is-not-defined)</sub>

**Comment by balazsorban44** [comment]
Confirmed, it's an upstream SWC bug, looking into it, thanks!

https://github.com/swc-project/swc/issues/7798
https://github.com/swc-project/swc/issues/7789

**Comment by j-mendez** [comment]
@balazsorban44 Hi, this issue is still appearing in production next.js@13.4.16 and took down my app since I overlooked and did not do a full E2E. I would not expect a simple patch change to cause a complete client-side application error like this. Is there a way we can prevent issues in this manner from occurring? I kinda always have to do a big E2E sweep after every patch or minor and it feels like there should be a balance of what is expected.

**Comment by joacub** [comment]
> @balazsorban44 Hi, this issue is still appearing in production next.js@13.4.16 and took down my app since I overlooked and did not do a full E2E. I would not expect a simple patch change to cause a complete client-side application error like this. Is there a way we can prevent issues in this manner from occurring? I kinda always have to do a big E2E sweep after every patch or minor and it feels like there should be a balance of what is expected.

Same thing happened to me…

**Comment by balazsorban44** [comment]
The fix was merged 20 hours ago, last release is from 2 days ago. It will be out soon on canary, followed up by a latest after that.

**Comment by joacub** [comment]
> The fix was merged 8 hours ago, (which is less than a day from reporting), last release is from 2 days ago.
> 
> I cut a new canary release, the fix is available in `13.4.17-canary.0`, please upgrade! A stable release will be out at most in a few days.
> 
> While unfortunate, bugs cannot be fully avoided. Generally you might want to lock your package versions to an exact version and run a full test on production critical pages to avoid issues.

This issue was not 2 days ago I repelentes this 2 weeks ago but yeah I agree bugs can not be fully avoided

**Comment by balazsorban44** [comment]
[13.4.17](https://github.com/vercel/next.js/releases/tag/v13.4.17) is out with the fix! Please upgrade `npm i next@latest`
